spec/sustainability.md

```diff
@@ -0,0 +1,41 @@
+# sustainability
+
+Our goal is to encourage and support the development of **Sustainable Web Apps**.
+
+For an app to be 'sustainable' we mean that on the one hand, it avoids unhealthy trends in the app development industry. It demonstrates:
+
+- No lock-in
+- No attention theft
+- Retention of control over personal data
+
+On the other hand, it does follow healthy trends which help users, and are coming to be expected:
+
+- Live collaboration
+- Linking of data between applications
+- Robust security
+
+We have arrived at these principles from analysis of the [use-cases](use-cases.md) (and, if we're honest, from the [prior art](design/prior-art.md) that has inspired us).
+
+## no lock-in
+
+No lock-in means that users are not tied to a specific platform or ecosystem. They have the freedom to migrate their data and switch to alternative solutions without losing valuable information or being dependent on a single provider. This promotes user autonomy and flexibility in choosing the tools that best fit their needs.
```

Comment:
> ... but it's in who-knows-what format, and something specifically designed to let you move to another similar service with no interruptions.

I think you've captured it rather well! I'll weave it into the text.

> ... _network effect_ lock-in, which is generally solved at the moment by federation.

I agree that this is important; I think it would be a stretch to say we're addressing it. I had been wondering how and where to point out the obvious objection to what we're working on: it's not going to work (to some intuitive definition) for social networks and other infinite-scroll apps.

Perhaps I'll add your point here, and explicitly de-scope it in the hypothesis.

Reply:
> I think it would be a stretch to say we're addressing it.

A bit of a stretch, yeah, especially at this point. But I do think it goes in the right direction. I don't know ActivityPub super closely, but I think (for example) the Gateway could participate in the Fediverse without too much more effort. Suppose, for instance, we used this setup to host a blog in the Fediverse. First, we'd need to have blog post resources which were HTTP-dereferenceable, which is exactly the kind of thing we've contemplated the Gateway hosting. Then, we'd need to do the pub-sub stuff to communicate ActivityStream `Activity`s when things are added or updated. That's outside the scope of what we've identified as the Gateway's role, but a similar server-y participant could do it, or both responsibilities could be taken over by a new Gateway-esque thing that's built completely outside of this spec.

So, we definitely don't get federation by implementing what we've described here, but I don't think it's totally unrelated. Whether it's worth saying anything about that in these docs, though, is a separate question.

Reply:
Meta-note: this discussion should be in a forum somewhere! I wonder if Sweap requires its own forum or if we should just use the [m-ld-spec Discussion](https://github.com/m-ld/m-ld-spec/discussions). Answers on a postcard.